torch/lib/c10d/reducer.cpp

```diff
@@ -51,7 +52,9 @@ Reducer::Reducer(
       bucket_bytes_cap_(bucket_bytes_cap),
       divFactor_(kUnsetDivFactor),
       comm_hook_(nullptr),
-      thread_local_state_(at::ThreadLocalState()) {
+      thread_local_state_(at::ThreadLocalState()),
+      ddp_debug_level_(parseDistDebugLevel()),
+      param_names_(std::move(paramNames)) {
```

Comment:
param_names_ can be used with unused_parameters_ when throwing error in line 504. This is the error where a lot of users hit and get confused. That would be great to give info for unused parameters there as well. It is better to give info including both of parameter index and parameter name.

Reply:
Great idea! To clarify, do you mean that in that error ("Expected to mark a parameter ready only once") we can log the index/name of the parameter being marked as ready twice?

Reply:
yes!